### DISCUSSION TITLE
TTController not finding IntelCPU Temperature Sensors

### DISCUSSION THREAD

**Discussion by EyoSama** [devcompl/TTController#181] [Configuration]
Hello Moshi,

First of all, nice piece of software! If only Thermaltake had programmers with your skills!

I did come across a problem. I was able to create a config file based on one of the examples you provided here and the info from my system. Everything appears to work really well except for using the CPU as a sensor. When I do a Report from TTController, I get the following sensors...

```
-------------------------------
Sensors
-------------------------------
»
»  Nuvoton NCT6798D:
»        Temperature:
»                Temperature #1 (/lpc/nct6798d/temperature/1): 27.5 °C
»                Temperature #2 (/lpc/nct6798d/temperature/2): 28.0 °C
»                Temperature #4 (/lpc/nct6798d/temperature/4): 21.0 °C
»                Temperature #5 (/lpc/nct6798d/temperature/5): 22.0 °C
»                Temperature #6 (/lpc/nct6798d/temperature/6): 60.0 °C
»
»  NVIDIA NVIDIA GeForce RTX 3090:
»        Temperature:
»                GPU Core (/gpu-nvidia/0/temperature/0): 41.0 °C
»
»  Samsung SSD 980 PRO 2TB:
»        Temperature:
»                Temperature (/nvme/1/temperature/0): 36.0 °C
»
»  Samsung SSD 860 EVO 4TB:
»        Temperature:
»                Temperature (/ssd/0/temperature/0): 21.0 °C
»
»  Samsung SSD 980 PRO 2TB:
»        Temperature:
»                Temperature (/nvme/2/temperature/0): 35.0 °C
»
-------------------------------
```

As you can see, there is no /intelcpu/0/temperature/8 or similar. I tried using that but could see in Debug that it could not be found. I then tried the Nuvoton sensors, but TTController was also not able to find "Temperature #1 (/lpc/nct6798d/temperature/1): 27.5 °C", even though it shows up in Report.

Do you have any idea what the problem could be?

I have 10 RiingQuad fans (2 controllers with 5 fans on each). i9-11900K and RTX3090.
The i9 temps do of course show up in HWINFO.

Any help would be greatly appreciated.

Cheers!

Eyo...

**Comment by MoshiMoshi0** [comment]
Probably LibreHardwareMonitor needs an update.
I'll update and then push a new build.

Also the  Nuvoton sensor is a Motherboard sensor, that wont be much use, but you can enable them to be used in the service with [MotherboardSensorsEnabled](https://moshimoshi0.github.io/TTController/config/service#motherboardsensorsenabled)

**Comment by MoshiMoshi0** [comment]
Please try this build: https://ci.appveyor.com/project/MoshiMoshi0/ttcontroller/builds/41247030/artifacts

**Reply by EyoSama** [reply]
Thank you very much, Moshi! The i9 sensors do now show up, but the service crashes when I try **Report** and also when I **Run in Console**. Here is the error from Run In Console...

>>
21:27:17.6486: ================================================================
21:27:17.6695: Initializing service, version "0.11.0-alpha.36+Branch.develop.Sha.14f0326fa52e4a2f559b48a551464e3b94585cda"
21:27:17.8853: Loading plugin assembly: TTController.Plugin.DpsgController [0.11.0.0]
21:27:17.8853: Loading plugin assembly: TTController.Plugin.Level20DeskController [0.11.0.0]
21:27:17.8873: Loading plugin assembly: TTController.Plugin.RiingController [0.11.0.0]
21:27:17.8873: Loading plugin assembly: TTController.Plugin.RiingPlusController [0.11.0.0]
21:27:17.8873: Loading plugin assembly: TTController.Plugin.RiingQuadController [0.11.0.0]
21:27:17.8873: Loading plugin assembly: TTController.Plugin.RiingTrioController [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.AuroraEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.BlendEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.BlinkEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.ColorSequenceEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.CopyColorEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.FlowEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.IpcEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.PingPongEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.PulseEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.RawEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.RazerConnectEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.RippleEffect [0.11.0.0]
21:27:17.9024: Loading plugin assembly: TTController.Plugin.SensorEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.SoundEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.SpectrumEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.StaticColorEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.WaveEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.ZoneEffect [0.11.0.0]
21:27:17.9203: Loading plugin assembly: TTController.Plugin.DelayPortModifier [0.11.0.0]
Service failed with unhandled exception:
System.TypeLoadException: Could not load type 'TTController.Common.Plugin.LedColorModifierBase`1' from assembly 'TTController.Common, Version=0.11.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at TTController.Service.Utils.PluginLoader.<Search>d__6.MoveNext() in C:\projects\ttcontroller\Source\TTController.Service\Utils\PluginLoader.cs:line 59
   at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at TTController.Service.Utils.PluginLoader.Load(String path, Type[] types) in C:\projects\ttcontroller\Source\TTController.Service\Utils\PluginLoader.cs:line 24
   at TTController.Service.Utils.PluginLoader.LoadAll(String path) in C:\projects\ttcontroller\Source\TTController.Service\Utils\PluginLoader.cs:line 21
   at TTController.Service.TTService.Initialize() in C:\projects\ttcontroller\Source\TTController.Service\TTService.cs:line 49
   at TTController.Service.Program.<>c.<MainMenu>b__5_1() in C:\projects\ttcontroller\Source\TTController.Service\Program.cs:line 49
Press any key to return to the menu...
<<

**Reply by EyoSama** [reply]
Ok, so I ended up deleting the folder after uninstalling the service. I only kept the Config.json and then copied this latest build. Once I re-installed and ran it again, everything seems to be working. Report works and so does Run in Console. I'm not sure what went wrong the first time.

I noticed the version number of this build is lower than your release beta one, will that be a problem? Also, I'm assuming that this was a Debug Compile, so will you be releasing a new Release build?

I am in awe of what you have done here. The Thermaltake fans are completely unusable without your service!! Great work!
